db/compaction/compaction_job.cc

```diff
@@ -877,11 +878,20 @@ void CompactionJob::ProcessKeyValueCompaction(SubcompactionState* sub_compact) {
 
   CompactionRangeDelAggregator range_del_agg(&cfd->internal_comparator(),
                                              existing_snapshots_);
+  ReadOptions read_options;
+  read_options.verify_checksums = true;
+  read_options.fill_cache = false;
```

Comment:
Should this be mentioned in HISTORY as a behavior change? Its a good thing that we'll avoid polluting the cache.

Reply:
Actually these options were set in the same way before this PR, just in `MakeInputIterator()`. I had to move the `ReadOptions` here so they can outlive the iterator returned by `MakeInputIterator()`, as its subiterators can now hold references.